core(legacy-javascript): add focus-visible polyfill

connorjclark opened the pull request:
This adds the first non-corejs polyfill. Inspired by https://github.com/GoogleChrome/web.dev/pull/9409

This polyfill can have an outsized impact relative to its size, so the audit in its current state does not do a good job of highlighting the issue. For example, it's passing because the KB is so little:

<img width="802" alt="image" src="https://user-images.githubusercontent.com/4071474/221021192-61bed0c3-b74d-4d10-a550-af8aeac5c5fb.png">

We can address this later, but it'll be either a huge change (task attribution using source maps...) or a hardcoded thing (force-fail audit when this polyfill is detected). Needs some discussion, but either way this is a good place to split the work and land something.

adamraine commented on core/test/audits/byte-efficiency/legacy-javascript-test.js:
```diff
@@ -251,6 +251,27 @@ describe('LegacyJavaScript audit', () => {
     ]);
   });
+
+  it('detects none-corejs modules from source maps', async () => {
```
```suggestion
  it('detects non-corejs modules from source maps', async () => {
```

adamraine commented on core/audits/byte-efficiency/legacy-javascript.js:
```diff
@@ -177,7 +177,12 @@ class LegacyJavascript extends ByteEfficiencyAudit {
 
   static getPolyfillData() {
-    return [
+    /** @type {Array<{name: string, modules: string[], corejs?: true}>} */
```
nit: I see we never do `corejs: false` so technically the type is correct, but why not just make the type `corejs?: boolean` in case we want to in the future?

adamraine commented on core/scripts/legacy-javascript/create-polyfill-size-estimation.js:
```diff
@@ -55,7 +55,8 @@ function getPolyfillDependencies() {
 
-  for (const {name, coreJs3Module} of LegacyJavascript.getPolyfillData()) {
+  for (const {name, modules} of LegacyJavascript.getPolyfillData().filter(d => d.corejs)) {
+    const coreJs3Module = modules[1];
```
It isn't super obvious as a consumer of `getPolyfillData` that `modules[0]` is always the coreJs2 module and `modules[1]` is the coreJs3 module.

WDYT about adding a `LegacyJavascript.getCorejsPolyfillData` function that does the `d.corejs` filter and maps to the old item structure?

connorjclark commented:
Updated - and also modified the polyfill size estimator graph to not be so specific to corejs. Just hard-coding the size of the non-corejs stuff for now, since it's only one thing.

connorjclark commented on core/audits/byte-efficiency/polyfill-graph-data.json:
```diff
@@ -1,54 +1,54 @@
 {
-  "moduleSizes": [498, 265, 277, 263, 453, 219, 216, 546, 339, 1608, 671, 1525, 420, 214, 504, 98, 524, 196, 268, 642, 204, 742, 618, 169, 394, 127, 433, 1473, 779, 239, 144, 182, 254, 77, 508, 124, 1388, 75, 133, 301, 362, 170, 1078, 182, 490, 195, 321, 316, 447, 551, 216, 284, 253, 17, 107, 295, 356, 345, 1939, 1596, 291, 139, 259, 1291, 179, 528, 174, 61, 326, 20, 444, 522, 104, 1945, 120, 194 ...
+  "moduleSizes": [11897, 498, 265, 277, 263, 453, 219, 216, 546, 339, 1608, 671, 1525, 420, 214, 504, 98, 524, 196, 268, 642, 204, 742, 618, 169, 394, 127, 433, 1473, 779, 239, 144, 182, 254, 77, 508, 124, 1388, 75, 133, 301, 362, 170, 1078, 182, 490, 195, 321, 316, 447, 551, 216, 284, 253, 17, 107, 295, 356, 345, 1939, 1596, 291, 139, 259, 1291, 179, 528, 174, 61, 326, 20, 444, 522, 104, 1945, 1 ...
```
Note how this first entry is the "common core-js" stuff, the same value as was in `baseSize`.
